tests/src/testklshell.cpp

```diff
@@ -151,6 +153,17 @@ static auto NonLinearKLShellLoadControlTR() {
   t.check(Dune::FloatCmp::eq(0.2087577577946809, maxDisp, 1e-6))
       << std::setprecision(16) << "The maximum displacement is " << maxDisp << "but it should be " << 0.2087577577946809
       << ". The difference is " << 0.2087577577946809 - maxDisp;
+
+  Eigen::MatrixXd M;
+  auto& fe = fes.front();
+  M.setZero(fe.size(), fe.size());
+  fe.calculateMatrixImpl<double>(req, MatrixAffordance::linearMass, M);
+
+  // All entries of diagonal should be greater than zero
+  t.check((M.diagonal().array() > 0).all());
```

Comment:
I think it is nice to have such [single element tests](https://github.com/ikarus-project/ikarus/blob/f1928da1cbeaa345c6bddd054afd65b1099a5404/tests/src/testnonlinearelasticity.hh#L242) also for all the elements. Can you also add it? Or may be it can be done in a separate PR

Reply:
There is one below, but there I would have needed to create the reqs, basis, ... all over again. But in general this is a good idea

Reply:
I can implement one for linearelastic at least